core/state_transition.go

```diff
@@ -204,19 +203,18 @@ func (st *StateTransition) buyGas(gasBailout bool) error {
 	}
 
 	// compute data fee for eip-4844 data blobs if any
-	dgBig := new(big.Int)
+	dgval := new(uint256.Int)
 	var dataGasUsed uint64
 	if st.evm.ChainRules().IsCancun {
 		dataGasUsed = st.dataGasUsed()
 		if st.evm.Context().ExcessDataGas == nil {
 			return fmt.Errorf("%w: sharding is active but ExcessDataGas is nil", ErrInternalFailure)
 		}
-		// TODO: Update the misc library to work with uint256 instead of math/big
-		dgBig.Mul(misc.GetDataGasPrice(st.evm.Context().ExcessDataGas), new(big.Int).SetUint64(dataGasUsed))
-	}
-	dgval, overflow := uint256.FromBig(dgBig)
-	if overflow {
-		return fmt.Errorf("%w: overflow converting datagas: %v", ErrInsufficientFunds, dgval)
+		dataGasPrice, err := misc.GetDataGasPrice(st.evm.Context().ExcessDataGas)
+		if err != nil {
+			return err
+		}
+		dgval.Mul(dataGasPrice, new(uint256.Int).SetUint64(dataGasUsed))
```

Comment:
use MulOverflow here to check for overflow.

Reply:
Oh yeah, thanks!